diff

This updates the interface to use pointer values, which is necessary
to support different architectures, i.e. 64-bit platforms.

Index: include/Scintilla.iface
===================================================================
--- include/Scintilla.iface	(revision 10437)
+++ include/Scintilla.iface	(working copy)
@@ -49,6 +49,7 @@
 ## Types:
 ##     void
 ##     int
+##     ptr -> a pointer (memory address)
 ##     bool -> integer, 1=true, 0=false
 ##     position -> integer position in a document
 ##     colour -> colour integer containing red, green and blue bytes.
@@ -908,11 +909,11 @@
 get int GetTextLength=2183(,)
 
 # Retrieve a pointer to a function that processes messages for this Scintilla.
-get int GetDirectFunction=2184(,)
+get ptr GetDirectFunction=2184(,)
 
 # Retrieve a pointer value to use as the first argument when calling
 # the function returned by GetDirectFunction.
-get int GetDirectPointer=2185(,)
+get ptr GetDirectPointer=2185(,)
 
 # Set to overtype (true) or insert mode.
 set void SetOvertype=2186(bool overtype,)
@@ -1386,10 +1387,10 @@
 set void SetViewEOL=2356(bool visible,)
 
 # Retrieve a pointer to the document object.
-get int GetDocPointer=2357(,)
+get ptr GetDocPointer=2357(,)
 
 # Change the document object used.
-set void SetDocPointer=2358(, int pointer)
+set void SetDocPointer=2358(, ptr pointer)
 
 # Set which document modification events are sent to the container.
 set void SetModEventMask=2359(int mask,)
@@ -1448,11 +1449,11 @@
 
 # Create a new document object.
 # Starts with reference count of 1 and not selected into editor.
-fun int CreateDocument=2375(,)
+fun ptr CreateDocument=2375(,)
 # Extend life of document.
-fun void AddRefDocument=2376(, int doc)
+fun void AddRefDocument=2376(, ptr doc)
 # Release a reference to the document, deleting document if it fades to black.
-fun void ReleaseDocument=2377(, int doc)
+fun void ReleaseDocument=2377(, ptr doc)
 
 # Get which document modification events are sent to the container.
 get int GetModEventMask=2378(,)
